resources/type-system/inference.md

```diff
@@ -696,6 +734,26 @@ constraint set `C` is define as follows:
 Note that the grounded constraint solution is a type, and hence may not contain
 occurences of the unknown type.
 
+#### Grounded constraint solution for a set of type variables
+
+The grounded constraint solution for a set of type variables `{X0, ..., Xn}`
+with respect to a constraint set `C`, with previous solution `{T0, ..., Tn}`, is
+defined to be the set of types `{U0, ..., Un}` such that:
+  - If `Ti` is known (that is, does not contain `_`), then `Ui = Ti`.  _(Note
+    that the upcoming "variance" feature will relax this rule so that it only
+    applies to type variables without an explicitly declared variance.)_
+  - Otherwise, if `Xi` does not have an explicit bound, then `Ui` is the
+    grounded constraint solution for the type variable `Xi` with respect to the
+    constraint set `C`.
+  - Otherwise, let `Bi` be the bound of `Xi`.  Then, let `Bi'` be the type
```

Comment:
Something is missing here.  The `Ti` are still type schemas, so `Bi` is a type schema.  Either we need to take a least (greatest?) closure here or.... something.